src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt

```diff
@@ -166,6 +178,120 @@ class TransportReplayChangesAction @Inject constructor(settings: Settings, trans
         return WriteReplicaResult(request, location, null, replicaShard, log)
     }
 
+    // fetches the index settings from the leader cluster and applies it to the local cluster's index settings.
+    private suspend fun syncRemoteSettings(leaderAlias: String, leaderIndex: String, followerIndex: String) {
```

Comment:
This function is being added mainly to be able to update mappings which use analyzer defined in the settings.
We can reduce the scope of this function to just check if there is any change in the `index.analysis.*` settings and then update that. This way we can remove most of the logic of creating the settings and closing the index.